onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -633,87 +707,80 @@ def list(self, request, *args, **kwargs):
-                        filter_.update({"id__in": submissions_ids})
-
-                    # Second scenario / Modify submissions based on list of ids
-                    elif payload.get("submissions_ids"):
-                        try:
-                            # Use int() to test if list of integers is valid.
-                            submissions_ids = payload.get("submissions_ids", [])
-                            or_ = {u"$or": [{u"_id": int(submission_id)} for submission_id in submissions_ids]}
-                            query.update(or_)
-                        except ValueError:
-                            raise ParseError(_("Invalid submissions ids: %(submissions_ids)s"
-                                               % {'submissions_ids': json.dumps(payload.get("submissions_ids"))}))
-
-                        filter_.update({"id__in": submissions_ids})
-                    # Third scenario / Modify all submissions in form, but confirmation param must be among payload
-                    elif payload.get("confirm", False) is not True:
-                        http_status = status.HTTP_400_BAD_REQUEST
-                        response = {"detail": _("No confirmations provided")}
+        # Second scenario / Get submissions based on list of ids
+        elif payload.get('submissions_ids'):
+            try:
+                # Use int() to test if list of integers is valid.
+                instances_ids = [int(submission_id)
+                                 for submission_id in payload.get(
+                                    'submissions_ids', [])]
+            except ValueError:
+                raise ParseError(
+                    _('Invalid submissions ids: %(submissions_ids)s'
+                      % {'submissions_ids': json.dumps(
+                        payload.get('submissions_ids'))}))
 
-                    # If everything is OK, submit data to DBs
-                    if http_status == status.HTTP_200_OK:
-                        # Update Postgres & Mongo
-                        updated_records_count = Instance.objects.\
-                            filter(**filter_).update(validation_status=new_validation_status)
-                        ParsedInstance.bulk_update_validation_statuses(query, new_validation_status)
-                        response = {"detail": _("{} submissions have been updated").format(updated_records_count)}
+        # Third scenario / get all submissions in form,
+        # but confirmation param must be among payload
+        elif payload.get('confirm', False) is not True:
+            raise NoConfirmationProvidedException()
 
-            return Response(response, http_status)
+        return instances_ids
+
+    def __validate_permission_on_bulk_action(self, request, permission):
+
+        xform = self.get_object()
+        if not request.user.has_perm(permission, xform):
+            raise PermissionDenied(_(u'You do not have validate permissions.'))
```

Comment:
This method is also called by `bulk_delete()`, not just when updating validation statuses